### PR TITLE
pipeline(ticdc): split dataflow in sinkNode

### LINKED ISSUE #5942: split dataflow in sinkNode

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #5942

### What is changed and how it works?
Split dataflow (emit & flush Redolog) in sinkNode instead of tableSink：
1. Decouple Redolog data stream from tableSink to facilitate [sink module refactoring](https://github.com/pingcap/tiflow/issues/5928). 
   Ref: https://github.com/pingcap/tiflow/pull/5929
4. Manage the Redolog data stream in sinkNode to facilitate [redoManager refactoring](https://github.com/pingcap/tiflow/issues/5920). 
   Ref: https://github.com/pingcap/tiflow/pull/5683

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

##### Will it cause performance regression or break compatibility?
No

##### Do you need to update user documentation, design documentation or monitoring documentation?
No

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/processor/pipeline/sink.go [hunk 1]
```diff
@@ -136,17 +143,37 @@ func (n *sinkNode) flushSink(ctx context.Context, resolved model.ResolvedTs) (er
...
-	if resolved.Ts > currentBarrierTs {
-		resolved = model.NewResolvedTs(currentBarrierTs)
-	}
 	if resolved.Ts > n.targetTs {
 		resolved = model.NewResolvedTs(n.targetTs)
 	}
+
+	if n.redoManager != nil && n.redoManager.Enabled() {
+		// redo log do not support batch resolve mode
+		if resolved.IsBatchMode() {
```
Comment (hi-rustin) [inline cdc/processor/pipeline/sink.go:154]: This is confusing to me, can you explain more? Also is the direct call to ResolvedMark here equivalent to the extra judgement you have now?

**Review by zhaoxinyu — APPROVED** [review]

File: cdc/processor/pipeline/sink.go [hunk 2]
```diff
@@ -136,17 +143,35 @@ func (n *sinkNode) flushSink(ctx context.Context, resolved model.ResolvedTs) (er
...
-	currentCheckpointTs := n.getCheckpointTs()
-	if resolved.Ts > currentBarrierTs {
-		resolved = model.NewResolvedTs(currentBarrierTs)
-	}
 	if resolved.Ts > n.targetTs {
 		resolved = model.NewResolvedTs(n.targetTs)
 	}
+
+	if n.redoManager != nil && n.redoManager.Enabled() {
+		// redo log do not support batch resolve mode
```
Comment (hi-rustin) [inline cdc/processor/pipeline/sink.go:153]: I think we could have explain it in more detail.

File: cdc/processor/pipeline/sink.go [hunk 3]
```diff
@@ -136,17 +143,35 @@ func (n *sinkNode) flushSink(ctx context.Context, resolved model.ResolvedTs) (er
...
+		resolved = model.NewResolvedTs(resolved.ResolvedMark())
+		err = n.redoManager.FlushLog(ctx, n.tableID, resolved.Ts)
+
+		redoTs := n.redoManager.GetMinResolvedTs()
```
Comment (hi-rustin) [inline cdc/processor/pipeline/sink.go:157]: Should we use this value to drive sink?

Comment (CharlesCheung96) [reply cdc/processor/pipeline/sink.go:157]: The resolvedTs used to drive Sink is only directly limited by barrierTs. In detail, the happens-before relation is:
`flushLog(resolvedTs from sorter) -> table sinkNode resolvedTs -> globalResolvedTs -> barrierTs`

Comment (hi-rustin) [reply cdc/processor/pipeline/sink.go:157]: Got it. Thanks!

**Review by hi-rustin — APPROVED** [review]
https://github.com/pingcap/tiflow/pull/5943#discussion_r904631065

rest LGTM

/hold

Please feel free to unhold this when you add the comments.

**Pull request merged**